src/main/java/com/thebois/models/beings/roles/BuilderRole.java

```diff
@@ -0,0 +1,13 @@
+package com.thebois.models.beings.roles;
+
+/**
+ * The builder constructs buildings.
+ */
+class BuilderRole extends AbstractRole {
+
+    @Override
+    public RoleType getType() {
+        return RoleType.BUILDER;
+    }
```

Comment:
I feel like having it in a method rather than a constructor is preferable as, generally speaking, methods are way more flexible than constructors. For example, you can't call the constructor multiple times. Therefore you can't ever change the `RoleType`. With a method, you can allow it to change dynamically. 

We probably don't want that, but it's just an example of the freedom we get by using methods rather than constructors. Is there a reason why constructors are better in this case?

Reply:
Nothing really, just felt weird looking at 6 classes all having the same method, and the only difference being what comes after RoleType. 

And constructors are usually often similar to each other and sometimes only differ in what arguments are sent in when they are subtypes. 

You are right though with saying methods are usually more flexible, so it might just be better to go with that and I don't mind either.

Reply:
Until someone comes up with a good reason to prefer constructors, I'd prefer us to go with methods for the above reason.